pkg/challenges/challenges.go

```diff
@@ -61,21 +63,87 @@ const (
 )
 
 type ChallengeResult struct {
-	Issuer    string
-	TypeVal   ChallengeType
-	PublicKey crypto.PublicKey
-	Value     string
+	Issuer  string
+	TypeVal ChallengeType
+	Value   string
 	// Extra information from the token that can be added to extensions.
 	AdditionalInfo map[AdditionalInfo]string
+
+	// subject or email from the id token. This must be the thing
+	// signed in the proof of possession!
+	subject string
+}
+
+func (cr *ChallengeResult) Name(context.Context) string {
+	return cr.subject
```

Comment:
I think this is an issue. Since subject and Value aren't the same, we're changing what's included in the certificate.

Why don't we stick with this being `cr.Value`, and stick with returning `ChallengeResult` for these internal methods so that you have access to `subject`?

Reply:
I'd kinda hope a test breaks too.

Reply:
Sorry, looked over the code again, I see we're not using `Name` where I thought. I guess that's another question, what do we need `Name` for?

Reply:
Yep, I agree that it didn't. I'm a little unclear about what Name is meant for, besides the proof of possession check. Is there a clearer function name than `Name` to distinguish between what's currently called `Value` (what goes into the certificate) and `subject` (The OIDC token subject/email)?

Reply:
If there isn't another planned use for `Name`, maybe we remove it? YAGNI unless it's going to be used elsewhere.

Reply:
Can we add a comment to `Value`, something like - "The identity that will be used as the subject alternative name of the certificate"

SG about changing in the next iteration.

Reply:
Yup absolutely. I'll clarify intended use on ChallengeResult right now and clarify the usage on the interface of Principal in an iteration